pkg/daemon/update.go

```diff
@@ -1796,6 +1796,13 @@ func (dn *Daemon) atomicallyWriteSSHKey(keys string) error {
 	// Once Users are supported fully this should be writing to PasswdUser.HomeDir
 	glog.Infof("Writing SSHKeys at %q", authKeyPath)
 
+	// Creating coreUserSSHPath in advance if it doesn't exist in order to ensure it is owned by core user
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=2107113
+	if _, err := os.Stat(coreUserSSHPath); os.IsNotExist(err) {
+		os.MkdirAll(coreUserSSHPath, os.FileMode(0o700))
+		os.Chown(coreUserSSHPath, uid, gid)
+	}
```

Comment:
This introduces a race condition - if we're interrupted between creating the directory and calling `chown()`, the directory will remain owned by root forever.

More generally, privileged code operating on directories owned by unprivileged users is full of *huge* traps.  For example, assuming the `core` user has sudo permissions removed, this might be subject to symlink attacks.

Both of these issues can be fixed by switching to the core user temporarily.  While it's possible to use low-level kernel APIs for this, in this case I'd say it's a lot simpler to fork off this as a subprocess: `runuser -u core -- mkdir -m 0700 -p ~/.ssh`.

Reply:
Really this applies anywhere in the MCO we want to write files that are in directories owned by lesser privileged users - but the core user ssh keys are probably the only case.

Reply:
I have changed the code with `runuser`. Now testing...